File: src/main/java/social/alone/server/location/LocationRepository.java
package social.alone.server.location;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;

import java.util.Optional;

public interface LocationRepository
        extends JpaRepository<Location, Long>, QuerydslPredicateExecutor<Location> {

  Optional<Location> findByLongitudeAndLatitudeAndName(double longitude, double latitude, String name);
}

File: src/main/kotlin/social/alone/server/location/LocationRepository.kt
package social.alone.server.location

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor

import java.util.Optional

interface LocationRepository : JpaRepository<Location, Long>, QuerydslPredicateExecutor<Location> {

    fun findByLongitudeAndLatitudeAndName(longitude: Double, latitude: Double, name: String): Optional<Location>
}
